File: Scripts/laptop/function-composition/c++/function-composition.cpp
#include <iostream>
#include <cmath>
#include <ext/functional>

int main() {
 volatile double r = 0; 

 for (int i = 0; i < 1000000000; ++i) {
     r = __gnu_cxx::compose1(std::ptr_fun(::sin), std::ptr_fun(::asin))(i);
  }
  return 0;
}

File: Scripts/laptop/function-composition/php/function-composition.php
<?php
function compose($f, $g) {
  return function($x) use ($f, $g) { return $f($g($x)); };
}

$results = 0;

for ($i = 0 ; $i < 1000000000; $i += 1) {
	$sin_asin = compose('sin', 'asin');
	$results = $sin_asin(0.5);
}

?>
